docs/source/usage.rst

```diff
@@ -432,14 +432,19 @@ The following example falls back to default passwords when ``DB_PASSWORD`` is no
 Decoding strings with interpolations
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Strings may be converted using ``oc.decode``:
+With ``oc.decode``, strings can be converted into their corresponding data types using the OmegaConf grammar.
+This grammar recognizes typical data types like ``bool``, ``int``, ``float``, ``dict`` and ``list``,
+e.g. ``"true"``, ``"1"``, ``"1e-3"``, ``"{a: b}"``, ``"[a, b, c]"``.
+It will also resolve interpolations like ``"${foo}"``, returning the corresponding value of the node.
 
-- Primitive values (e.g., ``"true"``, ``"1"``, ``"1e-3"``) are automatically converted to their corresponding type (bool, int, float)
-- Dictionaries and lists (e.g., ``"{a: b}"``, ``"[a, b, c]"``)  are returned as transient config nodes (DictConfig and ListConfig)
-- Interpolations (e.g., ``"${foo}"``) are automatically resolved
-- ``None`` is the only valid non-string input to ``oc.decode`` (returning ``None`` in that case)
+Note that:
 
-This can be useful for instance to parse environment variables:
+- When providing as input to ``oc.decode`` a string that is meant to be decoded into another string, in general
+  the input string should be quoted (since only a subset of characters are allowed by the grammar in unquoted
+  strings). For instance, a proper string interpolation could be: ``"'Hi! My name is: ${name}'"`` (with extra quotes).
+- ``None`` (written as ``null`` in the grammar) is the only valid non-string input to ``oc.decode`` (returning ``None`` in that case)
```

Comment:
Right, I can see how this can be confusing. I'm talking about the string being provided as input to `oc.decode`. When you write:
```python
"c": "${oc.deprecated:a, 'message: ${msg}'}",
```
The string provided to `oc.deprecated` isn't `"message: ${msg}"`, it is `"message: E.T. call home"`.

In general strings provided as input to `oc.decode` are much more likely to come from resolvers than to be written explicitly in the config (because if you can write it explicitly then there is no need for `oc.decode`). That's why my example also uses `oc.env`.

I'll need to think a bit more about it to try and figure out how to make it understandable in the doc. Suggestions welcome ;)

Reply:
I suggest we don't mention it.

Reply:
Also removed in 01ddb95